### ISSUE TITLE
Web URL not rendered in book page-footers on Windows

### ISSUE THREAD

**Issue by Hemken** [quarto-dev/quarto-cli#8737]
### Bug description

When rendering a book project on a Windows platform, when a page-footer includes a URL to a web site, Quarto errors out.  No output document is produced.

The same files, when rendered on a Linux platform, work as expected.

### Steps to reproduce

- `quarto create project book test-book`
- add to `_quarto.yml` (under `chapters:`)
  ````
  page-footer:
    left:
      - text: "Quarto"
        href: "https://quarto.org"
  ````
- render project on a Windows computer

### Expected behavior

This should render an HTML book, with the appropriate link in the page footer.

### Actual behavior

When rendered in Rstudio, the following result appears in the Background Jobs tab
````
ERROR: The specified path is invalid. (os error 161): stat 'Z:\test-book\https:\quarto.org'

Stack trace:
    at Object.statSync (ext:deno_fs/30_fs.js:363:7)
    at createWalkEntrySync (file:///C:/PROGRA~1/Quarto/bin/quarto.js:2305:23)
    at expandGlobSync (file:///C:/PROGRA~1/Quarto/bin/quarto.js:2439:25)
    at expandGlobSync.next (<anonymous>)
    at expandGlobs (file:///C:/PROGRA~1/Quarto/bin/quarto.js:5448:24)
    at resolveGlobs (file:///C:/PROGRA~1/Quarto/bin/quarto.js:5525:26)
    at resolvePathGlobs (file:///C:/PROGRA~1/Quarto/bin/quarto.js:5460:12)
    at projectInputFiles (file:///C:/PROGRA~1/Quarto/bin/quarto.js:73059:26)
    at projectContext (file:///C:/PROGRA~1/Quarto/bin/quarto.js:72803:44)
    at eventLoopTick (ext:core/01_core.js:183:11)
Error in `processx::run(quarto_bin, args, echo = TRUE)`:
! System command 'quarto.exe' failed
---
Exit status: 1
stdout & stderr: <printed>
---
Backtrace:
1. .rs.sourceWithProgress(script = "C:/Users/hemken/AppData/Local/Temp/RtmpkH…
2. base::eval(statements[[idx]], envir = globalenv())
3. base::eval(statements[[idx]], envir = globalenv())
4. quarto::quarto_render()
5. processx::run(quarto_bin, args, echo = TRUE)
6. processx:::throw(new_process_error(res, call = sys.call(), echo = echo, …
Execution halted
````

### Your environment

- IDE: RStudio 2023.09.1+494 "Desert Sunflower" Release (cd7011dce393115d3a7c3db799dda4b1c7e88711, 2023-10-16) for windows
- OS: Windows 10

### Quarto check output

Quarto 1.4.549
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.549
      Path: C:\Program Files\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: Installation From Path
      Path: C:\Program Files\texlive\2023\bin\windows
      Version: 2023

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

(|) Checking R installation...........Finished Rprofile.site
[>] Checking R installation...........OK
      Version: 4.3.2
      Path: C:/PROGRA~1/R/R-43~1.2
      LibPaths:
        - U:/R/4.3.2
        - C:/Program Files/R/R-4.3.2/site-library
        - C:/Program Files/R/R-4.3.2/library
      knitr: 1.45
      rmarkdown: 2.25

[>] Checking Knitr engine render......OK

**Comment by Hemken** [comment]
I should add that this seems to be specifically a problem with book projects, not single page website projects.

**Comment by mcanouil** [comment]
FWIW, on macOS, it works as expected.

**Comment by cderv** [comment]
This is unexpected that the error happens from `projectInputFiles`. We go through here: 
https://github.com/quarto-dev/quarto-cli/blob/d047195554ca875b420b99d98be5593810e3ee5e/src/project/project-context.ts#L762-L767

and it errors because `renderFiles = metadata?.project[kProjectRender];` contains the website 
````
> metadata.project.render
[
  "index.qmd",
  "intro.qmd",
  "summary.qmd",
  "references.qmd",
  "https://quarto.org",
]
````

I need to look how it goes on linux. It reminds me of issue with path detection between Linux and Windows for `https:://` 

This could end up there because `page-footer` may possibly take a .qmd file so we add in Render when detected, and we detect wrong on Windows.

**Comment by cderv** [comment]
Any `href` in `page-footer` will be considered a file to render
https://github.com/quarto-dev/quarto-cli/blob/d047195554ca875b420b99d98be5593810e3ee5e/src/project/types/book/book-config.ts#L250-L268

That is why it ended up in render list file. 

This list will go through `resolvePathGlobs` to be expanded to absolute path and this is where the issue happens on Windows. 

I believe this windows issue is just unveiling a hidden behavior in Linux: URL that gets in glob to be expanded in our function
https://github.com/quarto-dev/quarto-cli/blob/d047195554ca875b420b99d98be5593810e3ee5e/src/core/path.ts#L177-L190
 with `expandGlobSync` are just filtered out. 

### Behavior on Linux 

````bash
> import { expandGlobSync } from "https://deno.land/std@0.217.0/fs/mod.ts";
undefined
> const href = "https://quarto.org"
undefined
> for (file of expandGlobSync(href)) { console.log(file) }
undefined
````

### Behavior on Windows

````powershell
> import { expandGlobSync } from "https://deno.land/std@0.217.0/fs/mod.ts";
undefined
> const href = "https://quarto.org"
undefined
>  for (file of expandGlobSync(href)) { console.log(file) }
Uncaught Error: La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte. (os error 123): stat 'C:\Users\chris\AppData\Local\Temp\quarto\https:\quarto.org'
    at Object.statSync (ext:deno_fs/30_fs.js:425:3)
    at createWalkEntrySync (https://deno.land/std@0.217.0/fs/_create_walk_entry.ts:22:21)
    at expandGlobSync (https://deno.land/std@0.217.0/fs/expand_glob.ts:264:21)
    at expandGlobSync.next (<anonymous>)
    at <anonymous>:1:27
````

Deno ignores this because it handle error for file not found at https://deno.land/std@0.217.0/fs/expand_glob.ts?source=#L136
````typescript
  let fixedRootInfo: WalkEntry;
  try {
    fixedRootInfo = await createWalkEntry(fixedRoot);
  } catch (error) {
    return throwUnlessNotFound(error);
  }
````

So this could also be seen as a Deno bug. It is really to same issue of invalid path on windows, not being seen as invalid on Linux. Windows errors for invalid path before not found error. 

Previously we add this type of error with `existsSync()` so we added `safeExistsSync()`. 

Related issue is #4124 with fix at #4129 by checking for https path. 

So should probably be the same fix here for book with page footer file detection. 

However, I think we are impacted by this hidden behavior in several places we'll keep discovering
